canonicalwebteam/image_template/__init__.py

```diff
@@ -45,58 +61,88 @@ def image_template(
+    # #crop_parameter
     if fill:
         cloudinary_options.append("c_fill")
 
     if not url_parts.netloc:
         raise Exception("url must contain a hostname")
 
     std_def_cloudinary_options = cloudinary_options.copy()
-    hi_def_cloudinary_options = cloudinary_options.copy()
-
     std_def_cloudinary_options.append("w_" + str(width))
-
-    if height is not None:
-        std_def_cloudinary_options.append("h_" + str(height))
-
-    if hi_def:
-        hi_def_cloudinary_options.append("w_" + str(int(width) * 2))
-
-        if height is not None:
-            hi_def_cloudinary_options.append("h_" + str(int(height) * 2))
-
     std_def_cloudinary_attrs = ",".join(std_def_cloudinary_options)
-    hi_def_cloudinary_attrs = ",".join(hi_def_cloudinary_options)
 
     # Decode the URL first to prevent double encoding
     decoded_url = unquote(url)
     encoded_url = quote(decoded_url, safe="")
-
     image_src = (
         f"{cloudinary_url_base}/"
         f"{std_def_cloudinary_attrs}/"
         f"{encoded_url}"
     )
 
-    image_srcset = (
-        f"{cloudinary_url_base}/c_limit,"
-        f"{hi_def_cloudinary_attrs}/{encoded_url} 2x"
-    )
+    # Generate srcset values with optimized logic for real-world usage
+    # https://vanillaframework.io/docs/settings/breakpoint-settings
+    if srcset_widths is None:
+        srcset_widths = [460, 620, 1036, 1681]
+
+    width_int = int(width)
+    srcset = []
+
+    # Only generate srcset for images larger than 100px to avoid
+    # unnecessary overhead
+    if width_int > 100:
+        max_srcset_width = max(srcset_widths)
+        # When hi_def is enabled, allow slightly larger for high-DPI
+        # But cap at 2x to avoid excessive payload
+        if hi_def:
+            max_width_limit = min(width_int * 2, max_srcset_width)
+        else:
+            max_width_limit = min(width_int, max_srcset_width)
+
+        # Include widths that make sense for responsive design
+        def create_srcset_url(width, options):
+            width_options = options.copy()
+            width_options.append(f"w_{width}")
+            width_attrs = ",".join(width_options)
+            return (
+                f"{cloudinary_url_base}/{width_attrs}/"
+                f"{encoded_url} {width}w"
+            )
+
+        # Generate srcset entries for standard widths
+        filtered_widths = [w for w in srcset_widths if w <= max_width_limit]
+        srcset.extend(
+            create_srcset_url(w, cloudinary_options) for w in filtered_widths
+        )
+
+        # Add original width if needed
+        existing_widths = {int(w) for w in filtered_widths}
+        if width_int <= max_width_limit and width_int not in existing_widths:
+            srcset.append(create_srcset_url(width_int, cloudinary_options))
+
+    image_srcset = ", ".join(srcset)
+
+    try:
+        sizes_attr = sizes.format(width, width)
+    except (IndexError, KeyError):
+        sizes_attr = sizes
```

Comment:
Instead of assigning it to a `sizes_attrs`, can we just use the existing `sizes` variable?
```suggestion
    try:
        sizes = sizes.format(width, width)
    except (IndexError, KeyError):
        pass
```